Helper/Cart.php

```diff
@@ -2863,11 +2875,18 @@ public function getSaleRuleDiscounts($quote)
         } else {
             /* @var \Magento\SalesRule\Api\Data\RuleDiscountInterface $ruleDiscounts */
             $extensionSaleRuleDiscounts = $address->getExtensionAttributes()->getDiscounts();
+            $cartFixedRules = $address->getCartFixedRules();
             if ($extensionSaleRuleDiscounts && is_array($extensionSaleRuleDiscounts)) {
                 foreach ($extensionSaleRuleDiscounts as $value) {
                     /* @var \Magento\SalesRule\Api\Data\DiscountDataInterface $discountData */
                     $discountData = $value->getDiscountData();
-                    $saleRuleDiscountsDetails[$value->getRuleID()] = $discountData->getAmount();
+                    $salesRuleId = $value->getRuleID();
+                    if (!empty($cartFixedRules) && array_key_exists($salesRuleId, $cartFixedRules)) {
+                        $rule = $this->ruleRepository->getById($salesRuleId);
+                        $saleRuleDiscountsDetails[$salesRuleId] = $discountData->getAmount() + $rule->getDiscountAmount() - $cartFixedRules[$salesRuleId];
```

Comment:
Could you explain why you use this logic? That would be good if you can refer some Magento core logic here.  I'm having a little worried since the split discounts feature is causing a lot of bugs.

Reply:
@dumega Could you please answer here?

Reply:
@johnmccombs1  @vitaliyreznikov  When there are multiple "Fixed amount discount for whole cart" sale rule applied to cart, M2 core has specific logic (https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/SalesRule/Model/Rule/Action/Discount/CartFixed.php#L116 and https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/SalesRule/Model/Rule/Action/Discount/CartFixed.php#L156) to calculate the discount applied to shipping amount, as a result, the actual discount amount of sale rule does not equal to its original discount amount.

eg.
There are two "Fixed amount discount for whole cart"  sale rules A and B. the original discount amount of A is $5.06 while it is $6.06 for B, they are both applied to cart. In some contexts, although the cart subtotal and shipping amount are both larger than $12.12, the actual discount total of cart is $12.06 other than $12.12 (5.06+6.06).
When looking into the M2 core, I found that discount from each sale rule has two parts, one is applied to line items and another one is applied to shipping, we can get the amount of first part by **$discountData->getAmount()**, the amount  of second part by **$rule->getDiscountAmount() - $cartFixedRules[$salesRuleId]**
![image](https://user-images.githubusercontent.com/3198527/151290052-f046f560-c9fd-471c-9a3f-5930b2598898.png)